samples/transform_feedback_separated_2.html

```diff
@@ -198,150 +146,138 @@
-            gl.enableVertexAttribArray(SPAWNTIME_LOCATION);
-            gl.enableVertexAttribArray(LIFETIME_LOCATION);
-            gl.enableVertexAttribArray(ID_LOCATION);
+            return program;
         }
 
-        function emitParticles() {
+        function render() {
+
             var time = Date.now() - appStartTime;
 
-            var programEmit = programs[PROGRAM_EMIT];
+            // Set the viewport
+            gl.viewport(0, 0, canvas.width, canvas.height - 10);
+
+            // Clear color buffer
+            gl.clearColor(0.0, 0.0, 0.0, 1.0);
+            gl.clear(gl.COLOR_BUFFER_BIT);
 
             // Toggle source and destination VBO
-            var sourceVBO = particleVBOs[currentSourceIdx];
             var sourceVAO = particleVAOs[currentSourceIdx];
+            var sourceVBO = particleVBOs[currentSourceIdx];
 
             var destinationVBO = particleVBOs[(currentSourceIdx + 1) % 2];
 
             gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, transformFeedback);
 
-            gl.useProgram(programEmit);
-
-            setupVertexAttributes(sourceVAO, sourceVBO);
+            gl.useProgram(program);
+            gl.bindVertexArray(sourceVAO);
 
             // Set transform feedback buffer
-            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, destinationVBO[Particle.POSITION]);
-            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, destinationVBO[Particle.VELOCITY]);
-            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, destinationVBO[Particle.SPAWN_TIME]);
-            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, destinationVBO[Particle.LIFE_TIME]);
-
-            // Turn off rasterization - we are not drawing
-            gl.enable(gl.RASTERIZER_DISCARD);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, destinationVBO[POSITION_LOCATION]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 1, destinationVBO[VELOCITY_LOCATION]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 2, destinationVBO[SPAWNTIME_LOCATION]);
+            gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 3, destinationVBO[LIFETIME_LOCATION]);
 
             // Set uniforms
-            gl.uniform1f(emitTimeLocation, time);
-            gl.uniform2f(emitAccelerationLocation, 0.0, ACCELERATION);
+            gl.uniform1f(drawTimeLocation, time);
+            gl.uniform4f(drawColorLocation, 0.0, 1.0, 1.0, 1.0);
+            gl.uniform2f(drawAccelerationLocation, 0.0, ACCELERATION);
 
-            // Emit particles using transform feedback
+            // Draw particles using transform feedback
             gl.beginTransformFeedback(gl.POINTS);
+            gl.enable(gl.BLEND);
+            gl.blendFunc(gl.SRC_ALPHA, gl.ONE);
             gl.drawArrays(gl.POINTS, 0, NUM_PARTICLES);
             gl.endTransformFeedback();
 
             // Restore state
-            gl.disable(gl.RASTERIZER_DISCARD);
             gl.useProgram(null);

```

Comment:
I think after combining the two steps, when can delete these unbinding calls. (move the `gl.useProgram(program)` to somewhere before `requestAnimationFrame(render);`  since we have only one program for now)